app/assets/javascripts/osem-schedule.js

```diff
@@ -14,66 +17,99 @@ var Schedule = {
   },
   remove: function(element) {
     var e =  $("#" + element);
-    var event_schedule_id = e.attr("event_schedule_id");
-    if(event_schedule_id != null){
-      var my_url = url + '/' + event_schedule_id;
-      var success_callback = function(data) {
-        console.log(data);
-        e.attr("event_schedule_id", null);
-        e.appendTo($(".unscheduled-events"));
-        e.find(".schedule-event-delete-button").hide();
+    var event_id = e.attr("event_id");
+    if(e.attr("event_schedule_id")!= null){
+      events_to_remove[event_id] = {
+        event: e,
+        previous_parent: e.parent()
+      };
+    }
+    delete events_to_save[event_id];
+    var unscheduled = $(".unscheduled-events");
+    e.appendTo(unscheduled);
+    e.find(".schedule-event-delete-button").hide();
+  },
+  add: function (previous_parent, new_parent, event) {
+    var params = {
+      previous_parent: previous_parent,
+      new_parent: new_parent,
+      event: event
+    }
+    var event_id = event.attr("event_id");
+    events_to_save[event_id] = params;
+    delete events_to_remove[event_id];
```

Comment:
Why is it hard to understand?

I have two arrays, one for the events that must be saved and another one for those that must be removed. So if you add a event to the `events_to_save` array, it has to be deleted from `events_to_remove`, as you don't want to remove it anymore. I don't understand what is not clear.